neo/io/nestio.py

```diff
@@ -220,37 +225,46 @@ def __read_spiketrains(self, gdf_id_list, time_unit,
             self._get_conditions_and_sorting(id_column, time_column,
                                              gdf_id_list, t_start, t_stop)
 
-        data = self.avail_IOs['gdf'].get_columns(
-            column_ids=column_ids,
-            condition=condition,
-            condition_column=condition_column,
-            sorting_columns=sorting_column)
-
-        # create a list of SpikeTrains for all neuron IDs in gdf_id_list
-        # assign spike times to neuron IDs if id_column is given
-        if id_column is not None:
-            if (gdf_id_list == []) and id_column is not None:
-                gdf_id_list = np.unique(data[:, id_column])
-
-            spiketrain_list = []
-            for nid in gdf_id_list:
-                selected_ids = self._get_selected_ids(nid, id_column,
-                                                      time_column, t_start,
-                                                      t_stop, time_unit, data)
-                times = data[selected_ids[0]:selected_ids[1], time_column]
-                spiketrain_list.append(SpikeTrain(
-
-                    times, units=time_unit,
-                    t_start=t_start, t_stop=t_stop,
-                    id=nid, **args))
-
-        # if id_column is not given, all spike times are collected in one
-        #  spike train with id=None
-        else:
-            train = data[:, time_column]
-            spiketrain_list = [SpikeTrain(train, units=time_unit,
-                                          t_start=t_start, t_stop=t_stop,
-                                          id=None, **args)]
+        spiketrain_list = []
```

Comment:
The spiketrains are read here as `SpikeTrain` object, however, meanwhile, a `SpikeTrainList` object is probably more appropriate. Perhaps this would be a separate PR.

Reply:
Then again, maybe one could just
```suggestion
        spiketrain_list = SpikeTrainList
```

Reply:
The NestIO was written before SpikeTrainList even existed, so yes, we should use SpikeTrainList, since this is also the superior way to load many spiketrains (which simulations will create)